lib/logger/lib/logger.ex

```diff
@@ -881,7 +881,10 @@ defmodule Logger do
   defguardp is_msg(msg) when is_binary(msg) or is_list(msg) or is_map(msg)
 
   @doc false
-  def __do_log__(level, fun, metadata) when is_function(fun, 0) and is_map(metadata) do
+  def __do_log__(level, fun, metadata, location \\ %{})
```

Comment:
Can you make the location a required argument and pass it as first argument (or right before metadata)? It would reflect that it actually comes before metadata (i.e. lower precedence).